torch/fx/experimental/proxy_tensor.py

```diff
@@ -1430,9 +1430,11 @@ def __torch_function__(
                 torch.amp.autocast_mode._exit_autocast,
             ]:
                 node.meta["val"] = None
+            # For autocast, the python APIs run so we don't have to run them again 
+            # here. 
+            if func is torch._C._set_grad_enabled:
```

Comment:
cc: @zou3519, @bdhirsh This is behavioral change from before.

Reply:
So, at the very least, pre-dispatch tracing should not change the state after it runs. If there is only one single torch.set_grad_enabled(False) call in the graph, after one does the pre-dispatch trace, it should not be off.

Reply:
Yep i wrapped it in ctx manager under export.